pyfr/plugins/pseudostats.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.plugins.base import BasePlugin, init_csv
+
+
+class PseudoStatsPlugin(BasePlugin):
+    name = 'pseudostats'
+    systems = ['*']
+
+    def __init__(self, intg, cfgsect, prefix):
+        super().__init__(intg, cfgsect, prefix)
+
+        if intg.formulation != 'dual':
+            raise RuntimeError('Formulation {0} not supported by plugin {1}'
+                               .format(intg.formulation, self.name))
+
+        self.flushsteps = self.cfg.getint(self.cfgsect, 'flushsteps', 500)
+
+        self.count = 0
+        self.stats = []
+        self.tprev = intg.tcurr
+
+        fvars = ','.join(intg.system.elementscls.privarmap[self.ndims])
+
+        # MPI info
+        comm, rank, root = get_comm_rank_root()
+
+        # The root rank needs to open the output file
+        if rank == root:
+            self.outf = init_csv(self.cfg, cfgsect,
+                                 'total_niter,t,pseudo-iter,' + fvars)
+        else:
+            self.outf = None
+
+    def __call__(self, intg):
+        # Process the sequence of pseudo residuals
+        for i, (iternr, pseudo_resid) in enumerate(intg.pseudostepinfo,
+                                                   start=self.count):
+            self.stats.append((iternr + self.count + 1, self.tprev, iternr + 1)
+                              + tuple(pseudo_resid))
+
+        # Update the total step count and save the current time
+        self.count += intg.pseudostepinfo[-1][0] + 1
```

Comment:
Residuals are computed and stats accumulated only after pseudo-niters-min. I want one of columns of the file to be the total number of iterations.